server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RuleActivatorContextFactory.java

```diff
@@ -73,7 +66,7 @@ private RuleActivatorContext create(RuleKey ruleKey, DbSession session, RuleActi
   }
 
   private RuleDto initRule(RuleKey ruleKey, RuleActivatorContext context, DbSession dbSession) {
-    Optional<RuleDto> rule = db.ruleDao().selectByKey(dbSession, ruleKey);
+    Optional<RuleDto> rule = getRule(dbSession, ruleKey);
     checkRequest(rule.isPresent(), "Rule not found: %s", ruleKey);
     context.setRule(rule.get());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Call "rule.isPresent()" before accessing the value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)